x-pack/plugins/security/server/config.ts

```diff
@@ -97,6 +97,7 @@ const providersConfigSchema = schema.object(
           ...getCommonProviderSchemaProperties(),
           realm: schema.string(),
           maxRedirectURLSize: schema.byteSize({ defaultValue: '2kb' }),
+          useRelayStateDeepLink: schema.boolean({ defaultValue: false }),
```

Comment:
__note:__ naming is hard, happy to accept any suggestions!

Reply:
This name makes sense to me.

My only "concern" was that we'll discover another use for `RelayState` unrelated to deep linking...that would cause us to either deprecate this, or introduce a new option which is mutually-exclusive to this one. Based on what I'm seeing around the 'net, it doesn't seem like `RelayState` is used for anything else but deep linking, so I'm good with what you have.

Reply:
> My only "concern" was that we'll discover another use for RelayState unrelated to deep linking..

It's hardly ever used for anything else in SAML tbh